### PR TITLE
fix(sctpd): Fix sock-addr array memory allocation

### LINKED ISSUE #11540: grpcpp sync crash with 1.7 and ipv6 enabled
### Your Environment

- **Version:** 1.7.0-1644523313-a451d376
- **Affected Component:** Access Gateway
- **Affected Subcomponent:** AGW service
- **Deployment Environment:** bare-metal (AGW)

### Describe the Issue
observed on phy-u6
with ipv6 feature enabled from orc8r config. we tried run few automated test cases from spirent side.

UE ipv6 address allocation with GTP transport using ipv6.
just when this test starts we see grpcpp_sync crash is observed on magma system and all magma process gets restarted.

we have attached core included with syslog, mme logs.
[mme.log](https://github.com/magma/magma/files/8052043/mme.log)
[core-1644617057-grpcpp_sync_ser-357646.gz](https://github.com/magma/magma/files/8052044/core-1644617057-grpcpp_sync_ser-357646.gz)
[syslog.zip](https://github.com/magma/magma/files/8052047/syslog.zip)

### PR DESCRIPTION
**Pull request by pshelar**
Fix typo where memory size is calculated for obj pointer rather
than actual object.

Signed-off-by: Pravin B Shelar <pbshelar@fb.com>

## Test Plan
`make test`
Ran bare metal attach test with IPv4 and IPv6 address, that is crashing sctpd on current master.

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

**Comment by ssanadhya** [comment]
@pshelar , could you please share a test plan?

**Review by ssanadhya — APPROVED** [review]
LGTM

**Review by ulaskozat — APPROVED** [review]
LGTM

File: lte/gateway/c/sctpd/src/util.cpp [hunk 1]
```diff
@@ -111,20 +111,22 @@ int set_sctp_opts(const int sd, const uint16_t instreams,
...
   struct sockaddr_in* ipv4_addr;
   struct sockaddr_in6* ipv6_addr;
 
   auto num_ipv4_addrs = req->ipv4_addrs_size();
   auto num_ipv6_addrs = req->ipv6_addrs_size();
   *num_addrs = num_ipv4_addrs + num_ipv6_addrs;
 
-  *addrs = (struct sockaddr*)calloc(*num_addrs, sizeof(struct sockaddr*));
-  if (*addrs == NULL) return -1;
+  addrs = (struct sockaddr*)calloc(*num_addrs, sizeof(struct sockaddr));
```
Comment (ulaskozat) [inline lte/gateway/c/sctpd/src/util.cpp]: for `sockadd_in6` this is still not large enough.

**Review by ulaskozat — CHANGES_REQUESTED** [review]
The size of `struct sockaddr` is not large enough to hold `struct sockaddr_in6`.

File: lte/gateway/c/sctpd/src/util.cpp (hunk 1 above)
Comment (pshelar) [reply lte/gateway/c/sctpd/src/util.cpp]: good find.

**Review by ulaskozat — APPROVED** [review]
LGTM. Do you want to add some unit tests here? :)

**Comment by pshelar** [comment]
> LGTM. Do you want to add some unit tests here? :)

I looked at existing sctp unit test framework, It will need extensive changes to add support to validate sctp listeners. I will create separate issue to track this unit test improvement.

**Pull request merged**